Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -75,9 +82,37 @@ public function handle(CommandInterface $command): CommandResult
         // publishing themselves
         $eventsToPublish = $this->commandBus->handle($command, $this);
 
+        // add "initiatingUserId" and "initiatingTimestamp" metadata to all events
+        // TODO: cleanup
+        $eventsToPublish = new EventsToPublish(
+            $eventsToPublish->streamName,
+            Events::fromArray($eventsToPublish->events->map(function (EventInterface|DecoratedEvent $event) {
+                $metadata = $event instanceof DecoratedEvent ? $event->eventMetadata->value : [];
+                $metadata['initiatingUserId'] = $this->initiatingUserId->value;
+                $metadata['initiatingTimestamp'] = (new \DateTimeImmutable())->format(\DateTimeInterface::ATOM);
```

Comment:
Consider already existing metadata and don't override (= copy user id & timestamp from existing events)

Reply:
TODO: Use single DateTime instance

Reply:
TODO: Add doc comment